docs/source/notes/autograd.rst

```diff
@@ -3,119 +3,398 @@
 Autograd mechanics
 ==================
 
-This note will present an overview of how autograd works and records the
-operations. It's not strictly necessary to understand all this, but we recommend
-getting familiar with it, as it will help you write more efficient, cleaner
-programs, and can aid you in debugging.
+This note will present an overview of how autograd works. It's not strictly necessary
+to understand all this, but we recommend getting familiar with it, as it will help you
+write more efficient, cleaner programs, and can aid you in debugging.
 
-.. _excluding-subgraphs:
+TL-DR:
 
-Excluding subgraphs from backward
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+- Mathematical formulation: we compute vector-Jacobian products using the chain rule. We support inplace operations using hidden versioning of the Tensors.
```

Comment:
I mean, not really "support"; rather, we help you avoid nasty bugs from in-place operations affecting saved values for backwards by versioning the tensors

Reply:
nit: in-place